apps/meteor/client/views/room/composer/messageBox/MessageBox.tsx

```diff
@@ -117,6 +122,224 @@ const MessageBox = ({
 	const composerPlaceholder = useMessageBoxPlaceholder(t('Message'), room);
 
 	const [typing, setTyping] = useReducer(reducer, false);
+	const [filesToUpload, setFilesToUpload] = useState<File[]>([]);
+	const dispatchToastMessage = useToastMessageDispatch();
+	const maxFileSize = useSetting('FileUpload_MaxFileSize') as number;
+
+	function handleFilesToUpload(filesList: File[], resetFileInput?: () => void) {
+		setFilesToUpload((prevFiles) => {
+			let newFilesToUpload = [...prevFiles, ...filesList];
+			if (newFilesToUpload.length > 6) {
+				newFilesToUpload = newFilesToUpload.slice(0, 6);
+				dispatchToastMessage({
+					type: 'error',
+					message: "You can't upload more than 6 files at once. Only the first 6 files will be uploaded.",
+				});
+			}
+
+			const validFiles = newFilesToUpload.filter((queuedFile) => {
+				const { name, size } = queuedFile;
+
+				if (!name) {
+					dispatchToastMessage({
+						type: 'error',
+						message: t('error-the-field-is-required', { field: t('Name') }),
+					});
+					return false;
+				}
+
+				if (maxFileSize > -1 && (size || 0) > maxFileSize) {
+					dispatchToastMessage({
+						type: 'error',
+						message: `${t('File_exceeds_allowed_size_of_bytes', { size: fileSize(maxFileSize) })}`,
+					});
+					return false;
+				}
+
+				return true;
+			});
+
+			return validFiles;
+		});
+
+		resetFileInput?.();
+	}
+
+	const handleRemoveFile = (index: number) => {
+		const temp = [...filesToUpload];
+		temp.splice(index, 1);
+		setFilesToUpload(temp);
+	};
+
+	const getHeightAndWidthFromDataUrl = (dataURL: string): Promise<{ height: number; width: number }> => {
+		return new Promise((resolve) => {
+			const img = new Image();
+			img.onload = () => {
+				resolve({
+					height: img.height,
+					width: img.width,
+				});
+			};
+			img.src = dataURL;
+		});
+	};
+	const uploadFile = (
```

Comment:
This whole file related code to make the upload should be moved to outside the composer, the composer should only handle the UI thing and call the upload function, all the rest, the upload, the attachments, etc, should be at the uploads file.

Reply:
done made new file for handling file upload